### ISSUE TITLE
Add support for python deaggregation of Java KPL aggregated records

### ISSUE THREAD

**Issue by redparham** [awslabs/kinesis-aggregation#81]
This project focuses mainly on supporting Kinesis (de)aggregation in the context of aws lambda. Would it be possible to make the deggregation code more generic so that it can be used to deaggregate records produced by Java KPL?

We have a use case where we produce using Java KPL and consume using python via boto3 [client.get_records](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kinesis.html#Kinesis.Client.get_records). There is no builtin support for processing these aggregated records from python: https://docs.aws.amazon.com/streams/latest/dev/kinesis-kpl-integration.html. The code in deaggregator.py iter_deaggregate_records(records) is coupled to kinesis/lambda but with some refactoring, I think it could satisfy my use case.

**Comment by IanMeyers** [comment]
Agreed. Separating the structure of the input event from the base64 decoding and deaggregating should be trivial to implement in the code with backward compatibility. Happy to take a PR on this.
